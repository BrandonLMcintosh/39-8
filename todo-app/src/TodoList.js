import { React, useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
	const INITIAL_STATE = [];
	const [items, setItems] = useState(INITIAL_STATE);

	const addItem = (itemObj) => {
		const newItemList = [...items, itemObj];
		setItems(newItemList);
	};

	const removeItem = (id) => {
		const newItemList = items.filter((item) => item.id !== id);
		setItems(newItemList);
	};

	return (
		<div>
			<NewTodoForm addItem={addItem} />
			<ul>
				{items.map((item) => (
					<Todo text={item.text} key={item.id} id={item.id} removeItem={removeItem} />
				))}
			</ul>
		</div>
	);
};

export default TodoList;
