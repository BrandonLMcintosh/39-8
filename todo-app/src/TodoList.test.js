import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import TodoList from "./TodoList";

it("renders without crashing", () => {
	render(<TodoList />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<TodoList />);
	expect(asFragment()).toMatchSnapshot();
});

it("should add new item", () => {
	const { queryByText, getByLabelText } = render(<TodoList />);
	const input = getByLabelText("text:");
	const button = queryByText("Add!");
	expect(queryByText("test1")).not.toBeInTheDocument();
	fireEvent.change(input, { target: { value: "test1" } });
	fireEvent.click(button);
	expect(queryByText("test1")).toBeInTheDocument();
});

it("should remove item", () => {
	const { queryByText, getByLabelText } = render(<TodoList />);
	const input = getByLabelText("text:");
	const addButton = queryByText("Add!");
	fireEvent.change(input, { target: { value: "test1" } });
	fireEvent.click(addButton);
	const button = queryByText("X");
	expect(queryByText("test1")).toBeInTheDocument();
	fireEvent.click(button);
	expect(queryByText("test1")).not.toBeInTheDocument();
});
