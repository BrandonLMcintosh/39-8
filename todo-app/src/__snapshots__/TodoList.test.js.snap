// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="text"
      >
        text: 
        <input
          id="text"
          name="text"
          placeholder="enter todo here"
          type="text"
          value=""
        />
      </label>
      <button>
        Add!
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
